hoomd/System.cc

```diff
@@ -590,6 +590,8 @@ PDataFlags System::determineFlags(unsigned int tstep)
     if (m_integrator)
         flags = m_integrator->getRequestedPDataFlags();
 
+    flags.set(); // hack to test flags always set
```

Comment:
Is this our permanent solution here. If not should we label this with a TODO or issue?

Reply:
This is not a permanent solution. I still need to test the performance implications of this.

Reply:
Unfortunately, setting all flags on results in a performance loss of 10-20% on V100. Half of that comes from the extra cost of the virial in the pair potential. The other half comes from ComputeThermo calling 3 kernels instead of 1. We can mitigate the ComputeThermo overhead, but that will still leave us with a at least a 5-10% performance hit to leave the virial calculation on all the time.

We will need to keep the flag system, at least for virial computation and possible for energy. Perhaps we should provide an API for the user to set the flags for quantities they plan to request in the API? We can update the GSD and future logging output operations to request the flags in the steps when they are triggered.

Alternately, we could re-compute forces and virials when the user requests virials. However, this will not work for the pressure tensor computed by ComputeThermo - it relies on the net virial summed by Integrator.

Reply:
I think an integrator flag would probably be cleanest. We could document that it only overrides if `virial_flag == True` and `internal_flag == False` (i.e. NPT would still need pressure even if the flag on the python side was set to false). We just then need to document that the property is only available when the flag is set to true or the simulation requires that pressure be calculated (however that should not be relied upon).

We do need a way to say to the integrator, however, to only set the flag on certain timesteps to be able to quickly log the pressure, etc... We could make the python flag a trigger that can be called to determine when to set the flag (I know that it would have to be set one step before pressure was desired, but we could under the hood call `Trigger(timestep + 1)`).

The logger is fairly agnostic to what is doing the calculation by design and that would be difficult to change. I think the Trigger solution might be the best for flexibility, but may be overly complicated generally. We could duck type the setting to make `flag=True` mean `trigger.Periodic(1)`.

I am honestly not sure what the best solution here is.

Reply:
I opened #607 to remind us this needs to be done.

Reply:
The temporary hack has been removed for now.